wcfsetup/install/files/lib/command/attachment/CopyAttachments.class.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace wcf\command\attachment;
+
+use wcf\data\attachment\Attachment;
+use wcf\data\attachment\AttachmentEditor;
+use wcf\data\attachment\AttachmentList;
+use wcf\data\file\thumbnail\FileThumbnailList;
+use wcf\data\object\type\ObjectType;
+use wcf\data\object\type\ObjectTypeCache;
+use wcf\system\file\processor\FileProcessor;
+
+/**
+ * Copies attachments from one object to another.
+ * Returns an array of old attachmentIDs as keys and new attachmentIDs as values.
+ *
+ * @author Olaf Braun
+ * @copyright 2001-2025 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.3
+ */
+final class CopyAttachments
+{
+    public function __construct(
+        public readonly string $sourceObjectType,
+        public readonly int $sourceObjectID,
+        public readonly string $targetObjectType,
+        public readonly int $targetObjectID,
```

Comment:
```suggestion
        private readonly string $sourceObjectType,
        private readonly int $sourceObjectID,
        private readonly string $targetObjectType,
        private readonly int $targetObjectID,
```

There is no reason to make these properties public. Always set the lowest possible visibility, because this can be reversed later if needed without breaking compatibility.

At the same time this keeps the API surface as low as possible without impacting the functionality.